R/derive_flag_select_msrc.R

```diff
@@ -0,0 +1,267 @@
+#'
+#' @param flag_events Flag events
+#'
+#'   A list of `flag_event()` objects is expected. For each event the condition
+#'   (`condition` field) is evaluated in the source dataset referenced by the
+#'   `dataset_name` field. If it evaluates to `TRUE` at least once, the new
+#'   variable is set to `true_value`.
+#'
+#' @param source_datasets Source datasets
+#'
+#'   A named list of datasets is expected. The `dataset_name` field of
+#'   `flag_event()` refers to the dataset provided in the list.
+#'
+#' @param new_var New variable
+#'
+#'   The specified variable is added to the input dataset.
+#'
+#' @param true_value True value
+#'
+#'   The new variable (`new_var`) is set to the specified value for all by
+#'   groups for which at least for one source (`sources`) the condition
+#'   evaluates to `TRUE`.
+#'
+#'   The values of `true_value`, `false_value`, and `missing_value` must be of
+#'   the same type.
+#'
+#' @param false_value False value
+#'
+#'   The new variable (`new_var`) is set to the specified value for all by
+#'   groups for which at all sources (`sources`) the condition evaluates to
+#'   `FALSE` or `NA`.
+#'
+#'   The values of `true_value`, `false_value`, and `missing_value` must be of
+#'   the same type.
+#'
+#' @param missing_value Values used for missing information
+#'
+#'   The new variable is set to the specified value for all by groups without
+#'   observations in any of the sources (`sources`).
+#'
+#'   The values of `true_value`, `false_value`, and `missing_value` must be of
+#'   the same type.
+#'
+#' @inheritParams derive_var_merged_exist_flag
+#'
+#' @return The output dataset contains all observations and variables of the
+#'   input dataset and additionally the variable specified for `new_var`.
+#'
+#' @details
+#'
+#'   1. For each `flag_event()` object specified for `flag_events`: The
+#'   condition (`condition`) is evaluated in the dataset referenced by
+#'   `dataset_name`. If the `by_vars` field is specified the dataset is grouped
+#'   by the specified variables for evaluating the condition. If named elements
+#'   are used in `by_vars` like `by_vars = exprs(USUBJID, EXLNKID = ECLNKID)`,
+#'   the variables are renamed after the evaluation. If the `by_vars` element is
+#'   not specified, the observations are grouped by the variables specified for
+#'   the `by_vars` argument.
+#'
+#'   1. The new variable (`new_var`) is added to the input dataset and set to
+#'   the true value (`true_value`) if for the by group at least one condition
+#'   evaluates to `TRUE` in one of the sources. It is set to the false value
+#'   (`false_value`) if for the by group at least one observation exists and for
+#'   all observations the condition evaluates to `FALSE` or `NA`. Otherwise, it
+#'   is set to the missing value (`missing_value`).
+#'
+#' @seealso [flag_event()]
+#'
+#' @family der_gen
+#' @keywords der_gen
+#'
+#' @export
+#'
+#' @examples
+#' library(dplyr)
+#'
+#' adsl <- tribble(
```

Comment:
Could we give a little more context to these examples.  Is this procedure dataset (pr) saying the two subjects had cancer treatments??

Reply:
Comments added